pkg/istiodcert/worker.go

```diff
@@ -261,7 +272,7 @@ func makeDNSNamesFromRevisions(namespace string, istioRevisions []string) (strin
 			continue
 		}
 
-		dnsNames = append(dnsNames, fmt.Sprintf("istiod%s.%s.svc", revision, namespace))
+		dnsNames = append(dnsNames, fmt.Sprintf("istiod-%s.%s.svc", revision, namespace))
```

Comment:
Here's an example of Istio using this prefix. It's the only example I could find in their code. There are various other examples in their test code and YAML:
 * https://github.com/istio/istio/blob/aa777b26ee935bb7e3b943a4925b76ea7320494a/istioctl/pkg/multixds/gather.go#L272-L277
 *  https://github.com/search?q=repo%3Aistio%2Fistio+%22istiod-%25s%22&type=code